babeld/message.c

```diff
@@ -554,7 +554,7 @@ parse_packet(const unsigned char *from, struct interface *ifp,
                 }
                 have_router_id = 1;
             }
-            if(!have_router_id && message[2] != 0) {
+            if(metric < 0xFFFF && !have_router_id && message[2] != 0) {
```

Comment:
Would it be better here to define a constant rather than using a number directly in different places?